src/sentry/search/events/builder.py

```diff
@@ -1600,12 +1603,22 @@ def __init__(
         # Don't do any of the actions that would impact performance in anyway
         # Skips all indexer checks, and won't interact with clickhouse
         self.dry_run = dry_run
+        assert dataset is None or dataset in [Dataset.PerformanceMetrics, Dataset.Metrics]
         super().__init__(
-            # Dataset is always Metrics
-            Dataset.Metrics,
+            # TODO: defaulting to Metrics for now so I don't have to update incidents tests. Should be
+            # PerformanceMetrics
+            Dataset.Metrics if dataset is None else dataset,
             *args,
             **kwargs,
         )
+        if dataset is Dataset.PerformanceMetrics:
+            # Hack while granularity isn't being handled by snuba
+            granularity = {
+                60: 1,
+                3600: 2,
+                86400: 3,
+            }[self.granularity.granularity]
+            self.where.append(Condition(Column("granularity"), Op.EQ, granularity))
```

Comment:
Are these the only 3 granularities we allow here?

Reply:
yeah, we dropped 10s (on the Snuba side) for performance metrics because performance has no use case for that and it creates a ton of data.